PIITokenizer.ts

```diff
@@ -0,0 +1,47 @@
+
+
+import { DateSanitizer, GenderSanitizer, StringSanitizer } from "./Sanitizer";
+import { QGramExpander, DateExpander } from "./Expander";
+import { PrivacyPreservingTokenizer } from "./PrivacyPreservingTokenizer";
+
+export class PIITokenizer {
+  stringSanitizer: StringSanitizer;
+  dateSanitizer: DateSanitizer;
+  genderSanitizer: GenderSanitizer;
+
+  dateExpander: DateExpander;
+  qGramExpander: QGramExpander;
+
+  tokenizer: PrivacyPreservingTokenizer;
+
+  constructor(bloomFilterLength: number = 200, numberOfHashFunctions: number = 20, privacyBudget: number = 3.0) {
+    this.stringSanitizer = new StringSanitizer();
+    this.dateSanitizer = new DateSanitizer();
+    this.genderSanitizer = new GenderSanitizer();
+    
+    this.dateExpander = new DateExpander();
+    this.qGramExpander = new QGramExpander();
+  
+    this.tokenizer = new PrivacyPreservingTokenizer(bloomFilterLength, numberOfHashFunctions, privacyBudget);
+  }
+
+  tokenize(firstName: string, lastName: string, dateOfBirth: Date, gender: string, other: string[] = []): Uint8Array {
+
+    // Sanitize fields
+    firstName = this.stringSanitizer.sanitize(firstName);
+    lastName = this.stringSanitizer.sanitize(lastName);
+    gender = this.genderSanitizer.sanitize(gender);
+
+    // Expand fields
+    let fields: string[] = [
+      ...this.qGramExpander.expand(firstName),
+      ...this.qGramExpander.expand(lastName),
+      ...this.dateExpander.expand(dateOfBirth),
+      gender,
```

Comment:
for consistency sake, it might be worth having a genderExpander that is a NoOp

Reply:
Ok, I have added the GenderExpander. In fact, it implements one small feature: given `M` or `F`, it expands it to `gender:M` and `gender:F`. I thought of doing it to prevent clashes from other string fields that may have `M` or `F` (e.g. a name or last name).

I know that we lowercase first and last name, but anyway, I thought it may be useful to better distinguish the two.
Let me know your thoughts on this.